src/lib/parameters/parameters.cpp

```diff
@@ -759,6 +836,118 @@ void param_set_used(param_t param)
 	}
 }
 
+int param_set_default_value(param_t param, const void *val)
+{
+	if (!handle_in_range(param)) {
+		PX4_ERR("set default value invalid param %d", param);
+		return PX4_ERROR;
+	}
+
+	if (val == nullptr) {
+		PX4_ERR("set default value invalid value");
+		return PX4_ERROR;
+	}
+
+	int result = PX4_ERROR;
+
+	param_lock_custom_defaults();
```

Comment:
Since this is now cpp. Why not  auto unlock pattern?

Reply:
Blind copypasta. Thanks, I'll update it.